src/hazardous/hash/blake2/blake2b.rs

```diff
@@ -148,6 +151,38 @@ impl Hasher {
     }
 }
 
+#[cfg_attr(docsrs, doc(cfg(feature = "safe_api")))]
+/// Example: custom digest size.
+/// ```rust
+/// use orion::{
+///     hazardous::hash::blake2::blake2b::{Blake2b, Digest},
+///     errors::UnknownCryptoError,
+/// };
+/// use std::io::{self, Read, Write};
+///
+/// // `reader` could also be a `File::open(...)?`.
+/// let mut reader = io::Cursor::new(b"some data");
+/// let mut hasher = Blake2b::new(64)?; // 512-bit hash
+/// std::io::copy(&mut reader, &mut hasher)?;
+///
+/// let digest: Digest = hasher.finalize()?;
+///
+/// # Ok::<(), Box<dyn std::error::Error>>(())
+/// ```
```

Comment:
This is a great and succinct example! One thing I'm missing is perhaps a mention of the error-handling - that `io::ErrorKind::Other` is the only error ever thrown, if updating the hashing state errors.

In addition, it's a good comment that was put within the `flush()` fn, but seems also like a great documentation candidate. Otherwise, users might wonder why we even have flush or attempt to use it. Not a big deal though.

Lastly, maybe it's worth mentioning shortly that `Write` simply wraps `update()`, so that a reader may tell that the init->update-finalize-reset logic applies here as well, as described in the module docs.